src/util.js

```diff
@@ -148,4 +159,28 @@ export default class Utils {
         error(err);
       });
   }
+
+  /**
+   * @param {string|number} input - a string to store in the buffer
```

Comment:
There are no common code paths based on this param, and `encoding` is only used for string. I think this should be split into separate functions.

Reply:
I'll split the code